lib/uplink/project.go

```diff
@@ -202,3 +207,20 @@ func (p *Project) OpenBucket(ctx context.Context, bucketName string, access *Enc
 func (p *Project) Close() error {
 	return nil
 }
+
+// CheckBucketAttribution Checks the bucket attribution
+func (p *Project) CheckBucketAttribution(ctx context.Context, bucketName string) (bool, error) {
+	partnerID, err := uuid.Parse(p.uplinkCfg.Volatile.PartnerID)
+	if err != nil {
+		return false, Error.Wrap(err)
+	}
+
+	err = p.metainfo.SetAttribution(ctx, bucketName, *partnerID)
```

Comment:
maybe we should only do this if the BucketInfo struct doesn't already have an attribution set?

in terms of RPC calls, the common case isn't the case where we need to set attribution, so maybe we say:
 1) get me bucket info
 2) bucket info says the bucket isn't attributed and i have a partner id set, so let's set it in the bucket info.

so that the next time
 1) get me bucket info
 2) bucket info says the bucket is already attributed, nothing to do

Reply:
All the bucket attribution logic is handled at the satellite side, and handles very similarly as suggested.

SetAttribution() on satellite side returns **nil** for:
- when no bucket or just bucket with no file segment exists (no entry exists in value attributionDB, with valid set partner ID )
- when already bucket attributed
SetAttribution() return **err** for:
- when already bucket with file segment exists 
- any value attribution error (like db read and/or Insert error occurs)

SetAttribution() on the lib uplink side, handles the satellite retuned err and nil with relevant log/debug info.

Reply:
Is your suggestion to store in the local meta bucket info the attribution ?

Reply:
Yes that was the intent, but removed it as it was creating confusion.